sdk/core/Azure.Core.Experimental/src/JsonData.cs

```diff
@@ -20,15 +18,204 @@ namespace Azure.Core.Dynamic
     /// A mutable representation of a JSON value.
     /// </summary>
-    [DebuggerDisplay("{DebuggerDisplay,nq}")]
-    [DebuggerTypeProxy(typeof(JsonDataDebuggerProxy))]
+    //[DebuggerDisplay("{DebuggerDisplay,nq}")]
+    //[DebuggerTypeProxy(typeof(JsonDataDebuggerProxy))]
     [JsonConverter(typeof(JsonConverter))]
     public partial class JsonData : DynamicData, IDynamicMetaObjectProvider, IEquatable<JsonData>
     {
-        private readonly JsonValueKind _kind;
-        private Dictionary<string, JsonData>? _objectRepresentation;
-        private List<JsonData>? _arrayRepresentation;
-        private object? _value;
+        private Memory<byte> _original;
+        private List<IJsonDataChange>? _changes;
+        private JsonElement _element;
+
+        internal JsonDataElement RootElement => new JsonDataElement(this, _element, "");
+
+        internal bool HasChanges => _changes != null && _changes.Count > 0;
+
+        internal bool TryGetChange<T>(string path, out T? value)
+        {
+            if (_changes == null)
+            {
+                value = default(T);
+                return false;
+            }
+
+            for (int i = _changes.Count - 1; i >= 0; i--)
+            {
+                var change = _changes[i];
+                if (change.Property == path)
+                {
+                    // TODO: does this do boxing?
+                    value = ((JsonDataChange<T>)change).Value;
+                    return true;
+                }
+            }
+
+            value = default;
+            return false;
+        }
+
+        internal bool TryGetChange<T>(ReadOnlySpan<byte> path, out T? value)
+        {
+            if (_changes == null)
+            {
+                value = default(T);
+                return false;
+            }
+
+            // TODO: re-enable optimizations
+            // (System.Text.Unicode.Utf8 isn't available to us here)
+            var pathStr = Encoding.UTF8.GetString(path.ToArray());
+
+            //Span<char> utf16 = stackalloc char[path.Length];
+            //OperationStatus status = System.Text.Unicode.Utf8.ToUtf16(path, utf16, out _, out int written, false, true);
+            //if (status != OperationStatus.Done)
+            //{ throw new NotImplementedException(); } // TODO: needs to allocate from the pool
+            //utf16 = utf16.Slice(0, written);
+
+            for (int i = _changes.Count - 1; i >= 0; i--)
+            {
+                var change = _changes[i];
+                if (change.Property == pathStr)
+                //if (change.Property.AsSpan().SequenceEqual(utf16))
+                {
+                    // TODO: does this do boxing?
+                    value = ((JsonDataChange<T>)change).Value;
+                    return true;
+                }
+            }
+
+            value = default;
+            return false;
+        }
+
+        internal void Set<T>(string path, T value)
+        {
+            // TODO: why was this here?
```

Comment:
I think it's a mistake, i.e. I looked into my prototype and I cannot tell why added it.